integration/spec/features/v2/save_and_return_spec.rb

```diff
@@ -39,10 +41,43 @@
     check_optional_text(page.text)
     form.email_confirmation.set('email@email.com')
     check_validation_error_message("Check your email address is correct")
-    form.email_confirmation.set('email@example.com')
+    form.email_confirmation.set('fb-acceptance-tests@digital.justice.gov.uk')
     continue
     check_optional_text(page.text)
     expect(page.text).to include('Your form has been saved')
-    expect(page.text).to include('We have sent a one-off link to email@example.com')
+    expect(page.text).to include('We have sent a one-off link to fb-acceptance-tests@digital.justice.gov.uk')
+
+    resume_progress_email = get_resume_email('save-and-return-v2-acceptance-test')
+    resume_link = extract_link_from_email(resume_progress_email)
+    
+    visit resume_link
+
+    expect(page.text).to include('Continue with "save-and-return-v2-acceptance-test"')
+    form.resume_secret_answer.set('bar')
+    continue
+    check_validation_error_message('Your answer is incorrect. You have 3 attempts remaining.')
+    form.resume_secret_answer.set('foo')
+    continue
+    # expect(page.text).to include('You have sucessfuly retrieved your saved information.')
+    expect(page.text).to include(q1_answer)
+
+    visit resume_link
+
+    expect(page.text).to include('That link has already been used')
+  end
+
+  def get_resume_email(reference_number)
+    find_save_and_return_email(id: reference_number, expect_emails: nil).select do |email|
+      email.subject.include?('Resuming your application to')
+    end.sort_by { |email| email.raw.payload.headers[18].value }.last
```

Comment:
how do we sort the emails here? Is it by timestamp?

Reply:
Yeah there's a header called 'Date', should probably just look for that by key